environments/dotnet/ExecutorModule.cs

```diff
@@ -0,0 +1,65 @@
+using Fission.DotNetCore.Compiler;
+using System.Collections.Generic;
+using Nancy;
+using System.IO;
+using System;
+
+namespace Fission.DotNetCore
+{
+    public class ExecutorModule : NancyModule
+    {
+        private const string CODE_PATH = "/userfunc/user";
+
+        private static Function _userFunc;
+
+        public ExecutorModule()
+        {
+            Post("/specialize", args => Specialize());
+            Get("/", _ => Run());
+            Post("/", _ => Run());
+            Put("/", _ => Run());
+            Head("/", _ => Run());
+            Options("/", _ => Run());
+            Delete("/", _ => Run());
+
+        }
+
+        private object Specialize()
+        {
+            var errors = new List<string>();
+            if (File.Exists(CODE_PATH))
+            {
+                var code = File.ReadAllText(CODE_PATH);
+                _userFunc = FissionCompiler.Compile(code, out errors);
+                if (_userFunc == null)
+                {
+                    var errstr = string.Join(Environment.NewLine, errors);
+                    Console.WriteLine(errstr);
+                    var response = (Response)errstr;
+                    response.StatusCode = HttpStatusCode.InternalServerError;
+                    return response;
+
+                }
+                return null;
+            }
+            else
+            {
+                var response = (Response)"Unable to locate code";
+                response.StatusCode = HttpStatusCode.InternalServerError;
+                return response;
+            }
+        }
+
+        private object Run()
+        {
+            if (_userFunc == null)
+            {
+                var response = (Response)"Generic container: no requests supported";
+                response.StatusCode = HttpStatusCode.InternalServerError;
+                return response;
+            }
+            var args = ((DynamicDictionary)Request.Query).ToDictionary();
```

Comment:
Agreed that a context class is a better solution but I really do not like bleeding the Nancy API into the core implementation, it could be a problem later down the road if either we want to change hosting technology or if the Nancy guys decides to change their API.

Even if it might seem like a lot of extra plumbing I think the best way to proceed would be to create our own wrapper extracting the items from the request as we go along.

Thoughts?